src/types/mod.rs

```diff
@@ -419,6 +426,44 @@ pub struct AccountTransactionDetails {
     pub effects: AccountTransactionEffects,
 }
 
+impl AccountTransactionEffects {
+    pub fn tx_type(&self) -> Option<TransactionType> {
```

Comment:
IMO we should avoid using abbreviations such as `tx`, at least in the api
(pun intended)

Reply:
a doc comment would also be nice here